src/main/kotlin/org/rust/ide/inspections/RsSortImplTraitMethodsInspection.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.IntentionWrapper
+import com.intellij.codeInspection.ProblemsHolder
+import org.rust.ide.intentions.SortImplTraitMethodsIntention
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.RsVisitor
+
+class RsSortImplTraitMethodsInspection : RsLocalInspectionTool() {
```

Comment:
I think different methods order is not potential error so I prefer to drop inspection and leave the only intention
@matklad what do you think?

Reply:
I suggest that we make it a *weak warning* at least at first. If the large number of people complain, we can turn this into inspection, **but** by this time we'll educate people that this stuff is available. 

However, there's a high chance that most of the folks would be okay with the weak warning: you can always disable it in the settings. 

As a comparison, Kotlin's "replace a nice for-loop with an unreadable sidefectful sequence of high-order ops which allocate on every intermediate step" definitely has false positives, but it's a weak warning non the less because it advertises the feature.

Reply:
> we can turn this into inspection

You mean intention, don't you?

Hm, then let's drop intention and make this inspection enabled by default
I really think that we need only one of them because user always can suppress highlighting of inspection and use only quick fix and if we have both intention and inspection we get following situation
<img width="270" alt="screen shot 2018-01-25 at 02 25 13" src="https://user-images.githubusercontent.com/2539310/35362647-395fecb8-0177-11e8-8743-66c903d3dec7.png">

Reply:
Yep, I meant "let's have a thing with a squiggly line and, if user complain, downgrade it to lightbulb only thing".